external-api/src/helpers/cron.ts

```diff
@@ -0,0 +1,75 @@
+import dotenv from 'dotenv';
+import { CronJob } from 'cron';
+import { Estore } from '../interfaces';
+import { Category } from './types/estore';
+import { eStoreSiteCreationCron } from '../cron/eStoreSiteCreationJob.cron';
+
+dotenv.config();
+
+const { ESTORE_CRON_MANAGER_SCHEDULE, TZ } = process.env;
+const jobs = new Map();
+
+/**
+ * Executes a cron job based on the provided eStore data and category.
+ * @param eStoreData - The eStore data.
+ * @returns A promise that resolves when the cron job is completed.
+ */
+const onTick = async (eStoreData: Estore) => {
+  await eStoreSiteCreationCron(eStoreData);
+};
+
+/**
+ * Handles the completion of the eStore CRON for a specific deal category.
+ * @param eStoreData - The eStore data object.
+ * @param category - The category of the deal.
+ * @returns A promise that resolves when the completion is handled.
+ */
+const onComplete = (eStoreData: Estore, category: Category) => {
+  console.info('✅ eStore %s CRON has been completed successfully for deal %s', category, eStoreData.dealId);
+};
+
+/**
+ * Creates and starts a CRON job for eStore data.
+ * @param eStoreData - The eStore data.
+ * @param category - The category of the eStore data.
+ * @returns A promise that resolves when the CRON job is started.
+ */
+export const cron = (eStoreData: Estore, category: Category, kill: boolean = false): Promise<boolean> => {
+  const id = `estore_cron_${category}_${eStoreData.dealId.toString()}`;
+
+  // Stop the CRON job
+  if (kill) {
+    const job = jobs.get(id) as CronJob;
+
+    console.info('❌ eStore %s CRON %s has been stopped for deal %s.', category, id, eStoreData.dealIdentifier);
+
+    jobs.delete(id);
+    job.stop();
```

Comment:
If you have the time I'd probably split this job killing section into another function so that wherever we call cron() it's us creating a job and we explicitly call something else for killing a job